import psycopg2
from psycopg2 import Error

try:
    # Подключение к существующей базе данных
    connection = psycopg2.connect(
        user="postgres",  # Замените на ваши данные
        password="111",  # Замените на ваши данные
        host="localhost",  # Замените на ваши данные
        port="5432",  # Обычно 5432
        database="medical_db_upd"  # Замените на ваши данные
    )

    # Создание курсора для выполнения операций с базой данных
    cursor = connection.cursor()

    # Создание таблицы hospital
    cursor.execute("""
        CREATE TABLE hospital (
            hospital_id INT PRIMARY KEY,
            hospital_name VARCHAR(100) NOT NULL,
            bed_count INT
        );
    """)

    # Добавление столбца hospital_type в таблицу hospital
    cursor.execute('''
        ALTER TABLE hospital
        ADD COLUMN hospital_type VARCHAR(50);
    ''')

    # Вставка данных в таблицу hospital
    cursor.execute("""
        INSERT INTO hospital (hospital_id, hospital_name, bed_count, hospital_type)
        VALUES
            (1, 'Mayo Clinic', 200, 'General'),
            (2, 'Cleveland Clinic', 400, 'General'),
            (3, 'Johns Hopkins', 1000, 'Specialized'),
            (4, 'UCLA Medical Center', 1500, 'General');
    """)
    connection.commit()
    print("Таблица hospital создана и заполнена данными")

    # Создание таблицы doctor
    cursor.execute("""
        CREATE TABLE doctor (
            doctor_id INT PRIMARY KEY,
            doctor_name VARCHAR(100) NOT NULL,
            hospital_id INT NOT NULL,
            FOREIGN KEY (hospital_id) REFERENCES hospital(hospital_id),
            joining_date DATE NOT NULL,
            speciality VARCHAR(100) NOT NULL,
            salary INT NOT NULL,
            experience INT
        );
    """)

    # Вставка данных в таблицу doctor
    cursor.execute("""
        INSERT INTO doctor (doctor_id, doctor_name, hospital_id, joining_date, speciality, salary, experience)
        VALUES
            (101, 'David', 1, '2005-02-10', 'Pediatric', 40000, NULL),
            (102, 'Michael', 1, '2018-07-23', 'Oncologist', 20000, NULL),
            (103, 'Susan', 2, '2016-05-19', 'Gynecologist', 25000, NULL),
            (104, 'Robert', 2, '2017-12-28', 'Pediatric', 28000, NULL),
            (105, 'Linda', 3, '2004-06-04', 'Gynecologist', 42000, NULL),
            (106, 'William', 3, '2012-09-11', 'Dermatologist', 30000, NULL),
            (107, 'Richard', 4, '2014-08-21', 'Gynecologist', 32000, NULL),
            (108, 'Karen', 4, '2011-10-17', 'Radiologist', 30000, NULL),
            (109, 'James', 1, '2022-01-15', 'Cardiologist', 45000, 5),
            (110, 'Emily', 1, '2023-04-10', 'Orthopedic Surgeon', 50000, 3),
            (111, 'Olivia', 2, '2021-09-05', 'Neurologist', 42000, 4),
            (112, 'John', 2, '2024-02-18', 'Surgeon', 60000, 2),
            (113, 'Sophia', 3, '2022-07-30', 'Urologist', 38000, 6),
            (114, 'Daniel', 3, '2025-03-22', 'Pulmonologist', 47000, 1),
            (115, 'Isabella', 4, '2023-11-01', 'Pediatrician', 41000, 3),
            (116, 'Liam', 4, '2022-05-25', 'Dermatologist', 35000, 4),
            (117, 'Mia', 1, '2024-06-17', 'Gastroenterologist', 53000, 2),
            (118, 'Lucas', 2, '2023-01-12', 'Anesthesiologist', 46000, 3);
    """)
    connection.commit()
    print("Данные успешно вставлены в таблицу 'Hospital'")

except (Exception, Error) as error:
    print("Ошибка при подключении к PostgreSQL:", error)

finally:
    # Закрытие соединения (если оно было установлено)
    if connection:
        if cursor:
            cursor.close()
        connection.close()
        print("Соединение с PostgreSQL закрыто")
